features/vat-replacement/run.js

```diff
@@ -0,0 +1,235 @@
+                                id,
+                                key,
+                                description, 
+                                rates {
+                                    name, 
+                                    amount, 
+                                    country, 
+                                    includedInPrice,
+                                    id
+                                },
+                                version
+                            }
+                        }
+                    }`,
+        endpoint: 'taxCategories',
+        variables: {
+            limit: 500
+        }
+    }
+
+    try {
+        let taxCategories = []
+        for await (const batch of ctpClient.fetchGraphQlBatches(params)) {
+            await Promise.map(batch, async (taxCategory) => {
+                taxCategories.push(taxCategory)
+            }, { concurrency: 4 })
+        }
+        return taxCategories
+    } catch (e) {
+        console.log(e)
+    }
+}
+
+async function replaceTaxRate(ctpClient, taxCategoryId, updateJsonObj) {
+    try {
+
+        await ctpClient.update(ctpClient.builder.taxCategories, taxCategoryId,
+            updateJsonObj.version, updateJsonObj.actions)
+    } catch (e) {
+        console.log(e)
+    }
+}
+
+// Filter the tax category list to obtain a list of tax rates which has following criteria :
+// 1. country code = 'DE'
+// 2. tax rate = 19% or 7%
+function getGermanValidTaxRateList(taxCategories, taxRateIdToTaxCategoryMap) {
+    let germanTaxRateList = taxCategories
+        .flatMap(item => item.rates)
+        .filter(rate => rate.country ==='DE')
+    let validGermanTaxRateList = []
+    if (germanTaxRateList.length === 0) {
+        let errMsg  =   'No valid tax rate from Germany. There is nothing to be done in your project in ' +
+                        'respect to VAT.' + '\n'
+        console.error(errMsg)
+        return validGermanTaxRateList
+    }
+    let invalidGermanTaxRateList = germanTaxRateList.filter(rate =>
+        rate.amount !== vatConstant.TAX_RATE_STANDARD_OLD &&
+        rate.amount !== vatConstant.TAX_RATE_REDUCED_OLD)
+    validGermanTaxRateList = germanTaxRateList
+        .filter(rate => rate.amount===vatConstant.TAX_RATE_STANDARD_OLD ||
+            rate.amount === vatConstant.TAX_RATE_REDUCED_OLD)
+
+    const standardVATs = validGermanTaxRateList.filter(rate => rate.amount === vatConstant.TAX_RATE_STANDARD_OLD)
+    const reducedVATs = validGermanTaxRateList.filter(rate => rate.amount === vatConstant.TAX_RATE_REDUCED_OLD)
+    let errMsg  = null
+    if (invalidGermanTaxRateList.length>0) {
+        errMsg  = 'We are sorry, we would have to ask you to change the vat manually if applicable. ' +
+            'There seems to be a special case.' + '\n'
+        errMsg +=  buildTaxRateDraftJson(invalidGermanTaxRateList, taxRateIdToTaxCategoryMap)
+        console.error(errMsg)
+    }
+    if (standardVATs.length > 1) {
```

Comment:
Did you test this case?
```
if there is more than one taxRate for "DE" that is 19 or 7 percent display these taxRates and the respective taxCategories together with a message like 'We are sorry but there are several tax rates for "DE" with a percentage of \variable_with_respective_value\ percent. Please update the taxRate manually' -> no JSON or API call
```
In this case, the taxRates would be in `validGermanTaxRateList` since they're valid, but we should not update the rates. In this method, you will return `validGermanTaxRateList` which contains them, and then you would process. Is it like this?

Reply:
Yes. Tested. 

I have talked with Thomas and confirmed. All warnings have been given out in preview mode already. If customers keep going on and run update, it would proceed.